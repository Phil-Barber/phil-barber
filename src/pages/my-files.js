import React from 'react';
import { graphql } from 'gatsby';
import PageWrappert from '../components/pageWrapper';

const MyFiles = ({ data }) => {
  console.log(data);
  return (
    <PageWrappert>
      <div>
        <h1>My Site&apos;s Files</h1>
        <table>
          <thead>
            <tr>
              <th>relativePath</th>
              <th>prettySize</th>
              <th>extension</th>
              <th>birthTime</th>
            </tr>
          </thead>
          <tbody>
            {data.allFile.edges.map(({ node }, index) => (
              <tr key={index}>
                <td>{node.relativePath}</td>
                <td>{node.prettySize}</td>
                <td>{node.extension}</td>
                <td>{node.birthTime}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    </PageWrappert>
  );
};

export default MyFiles;

export const query = graphql`
  query {
    allFile {
      edges {
        node {
          relativePath
          prettySize
          extension
          birthTime(fromNow: true)
        }
      }
    }
  }
`;
