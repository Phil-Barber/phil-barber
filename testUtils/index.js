import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../src/style';
import '@testing-library/jest-dom/extend-expect';

const customRender = (ui, options, store) => {
  const AllTheProviders = ({ children }) => (
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  );
  return render(ui, { wrapper: AllTheProviders, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
