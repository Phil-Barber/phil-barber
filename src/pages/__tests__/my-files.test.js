import React from 'react';
import { render } from '../../../testUtils';
import { MyFiles } from '../my-files';
import { useStaticQuery } from 'gatsby';

describe('MyFiles', () => {
  useStaticQuery.mockImplementation(() => ({
    site: {
      siteMetadata: {
        title: 'Test title',
      },
    },
  }));

  it('Matches snapshot', () => {
    const { container } = render(
      <MyFiles
        data={{
          allFile: {
            edges: [
              {
                node: {
                  relativePath: './test',
                  prettySize: 10,
                  extension: '.js',
                  birthTime: '12:00pm',
                },
              },
              {
                node: {
                  relativePath: './test2',
                  prettySize: 20,
                  extension: '.js',
                  birthTime: '12:00pm',
                },
              },
            ],
          },
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
