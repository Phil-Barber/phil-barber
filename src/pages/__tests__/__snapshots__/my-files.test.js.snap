// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFiles Matches snapshot 1`] = `
<div>
  <div
    class="headroom-wrapper"
  >
    <div
      class="headroom headroom--unfixed"
      style="position: relative; top: 0px; left: 0px; right: 0px; z-index: 1; transform: translate3D(0, 0, 0);"
    >
      <div
        class="sc-AxjAm fEzfgO"
      >
        <a
          class="sc-AxhUy bILkkK"
          href="/"
        >
          <h1>
            Test title
          </h1>
        </a>
        <span
          class="sc-AxirZ gtVPNy"
          style="transform: rotate(0deg);"
        >
          &gt;
        </span>
      </div>
    </div>
  </div>
  <div
    class="sc-AxiKw jmqPVm"
  >
    <div
      class="sc-AxheI gPjnvY"
    >
      <div
        class="sc-AxmLO gZPaXN"
      >
        <div>
          <h1>
            My Site's Files
          </h1>
          <table>
            <thead>
              <tr>
                <th>
                  relativePath
                </th>
                <th>
                  prettySize
                </th>
                <th>
                  extension
                </th>
                <th>
                  birthTime
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  ./test
                </td>
                <td>
                  10
                </td>
                <td>
                  .js
                </td>
                <td>
                  12:00pm
                </td>
              </tr>
              <tr>
                <td>
                  ./test2
                </td>
                <td>
                  20
                </td>
                <td>
                  .js
                </td>
                <td>
                  12:00pm
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div
      class="sc-AxhCb jDuQaB"
    />
  </div>
</div>
`;
